alpha/apps/kaltura/lib/storage/kFileSyncUtils.class.php

```diff
@@ -2223,4 +2223,31 @@ public static function getFileSyncFullPath(FileSync $fileSync, $pathOnly = true)
 		}
 		return $fullPath;
 	}
+
+	public static function getFileSyncAndPathForFlavor($syncKey, $flavorAsset)
+	{
+		$path = '';
+		$parent_file_sync = null;
+		$preferredStorageId = serveFlavorAction::$preferredStorageId;
```

Comment:
Instead of accessing external value from utils, better pass it to the function as parameter.